Packages/com.chisel.editor/Chisel/Editor/Editor/MaterialBrowser/ChiselMaterialBrowserUtilities.cs

```diff
@@ -0,0 +1,242 @@
+        }
+
+        private static void PopulateUsedMaterials( ref List<ChiselMaterialBrowserTile> tiles, ref List<ChiselModel> models, string searchLabel, string searchText )
+        {
+            tiles.ForEach( e =>
+            {
+                e.Dispose();
+                e = null;
+            } );
+
+            tiles.Clear();
+
+            foreach( ChiselModel m in models )
+            {
+                MeshRenderer[] renderMeshes = m.generated.meshRenderers;
+
+                if( renderMeshes.Length > 0 )
+                    foreach( MeshRenderer mesh in renderMeshes )
+                    {
+                        foreach( Material mat in mesh.sharedMaterials )
+                        {
+                            if( mat != null )
+                                if( mat.name.Contains( searchText ) || MaterialContainsLabel( searchLabel, mat ) )
+                                {
+                                    AssetDatabase.TryGetGUIDAndLocalFileIdentifier( mat, out string guid, out long id );
+
+                                    tiles.Add( new ChiselMaterialBrowserTile( guid ) );
+                                }
+                        }
+                    }
+            }
+        }
+
+        private static bool MaterialContainsLabel( string label, Material material )
+        {
+            string[] labels = AssetDatabase.GetLabels( material );
+
+            if( labels.Length > 0 )
+                foreach( string l in labels )
+                {
+                    if( l.Contains( label ) ) return true;
+                }
+
+            return false;
+        }
+
+        public static void SelectMaterialInScene( string name )
+        {
+            MeshRenderer[] objects = Object.FindObjectsOfType<MeshRenderer>();
+
+            if( objects.Length > 0 )
+                foreach( MeshRenderer r in objects )
+                {
+                    if( r.sharedMaterials.Length > 0 )
+                        foreach( Material m in r.sharedMaterials )
+                        {
+                            if( m != null )
+                                if( m.name.Contains( name ) )
+                                {
+                                    EditorGUIUtility.PingObject( r.gameObject );
+                                    Selection.activeObject = r.gameObject;
+                                }
+                        }
+                }
+        }
+
+        private static MethodInfo m_GetAssetPreviewMethod;
+
+        public static Texture2D GetAssetPreviewFromGUID( string guid )
+        {
+            m_GetAssetPreviewMethod ??= typeof( AssetPreview ).GetMethod(
+                    "GetAssetPreviewFromGUID",
+                    BindingFlags.Static | BindingFlags.NonPublic | BindingFlags.InvokeMethod,
+                    null,
+                    new[] { typeof( string ) },
+                    null
+            );
```

Comment:
Take a look at ReflectionExtensions and the usages of the methods there.
you can probably do something like

public delegate Texture2D GetAssetPreviewFromGUID(string guid);
private static GetAssetPreviewFromGUID m_GetAssetPreviewMethod = typeof(AssetPreview ).CreateDelegate<GetAssetPreviewFromGUID>("GetAssetPreviewFromGUID");